terraso_backend/apps/auth/services.py

```diff
@@ -62,7 +62,22 @@ def sign_up_with_apple(self, authorization_code, first_name="", last_name=""):
 
         return self._persist_user(tokens.open_id.email, first_name=first_name, last_name=last_name)
 
+    def sign_up_with_microsoft(self, authorization_code):
+        provider = MicrosoftProvider()
+        tokens = provider.fetch_auth_tokens(authorization_code)
+        if not tokens.is_valid:
+            error_msg = f"Error fetching auth tokens: {tokens.error_description}"
+            logger.error(error_msg)
```

Comment:
question: should this be `logger.exception`?

Reply:
I think `error` is fine here, as no exception stacktrace needs to be printed.